elytron/src/main/java/org/wildfly/extension/elytron/AuthenticationClientDefinitions.java

```diff
@@ -317,7 +318,12 @@ static ResourceDefinition getAuthenticationClientDefinition() {
                         try {
                             CredentialSource cs = sourceSupplier.get();
                             if (cs != null) {
-                                return c.usePassword(cs.getCredential(PasswordCredential.class).getPassword());
+                                PasswordCredential passCredential = cs.getCredential(PasswordCredential.class);
+                                String alias = credentialReference.hasDefined(CredentialReference.ALIAS) ? credentialReference.get(CredentialReference.ALIAS).asString() : null;
+                                if (passCredential == null && alias != null && alias.length() > 0) {
+                                    throw ROOT_LOGGER.credentialDoesNotExist(alias, PasswordCredential.class.getName());
+                                }
+                                return c.usePassword(passCredential.getPassword());
```

Comment:
This can still NPE if alias == null, although perhaps it's not really possible for passCredential to be null if alias is null. (I haven't looked to see.)